libs/common-services/src/main/java/uk/gov/di/ipv/core/library/domain/Name.java

```diff
@@ -33,4 +35,13 @@ public String getFullName() {
 
         return nameBuilder.toString();
     }
+
+    @JsonIgnore
+    public Map<String, String> getFormattedName() {
+        Map<String, String> formattedName = new HashMap<>();
+        for (NameParts namePart : nameParts) {
+            formattedName.put(namePart.getType(), namePart.getValue());
+        }
+        return formattedName;
```

Comment:
I don't think this will work - you can have multiple name parts of the same type. Have a look at the examples linked on https://vocab.account.gov.uk/v1/classes/NameClass/ , e.g.:
```
{
  "nameParts": [
    {
      "value": "Ana",
      "type": "GivenName"
    },
    {
      "value": "Maria",
      "type": "GivenName"
    },
    {
      "value": "Carvalho",
      "type": "FamilyName"
    },
    {
      "value": "Sousa",
      "type": "FamilyName"
    }
  ]  
}
```

Reply:
TBH I would probably suggest that you just pass `nameParts` directly to the frontend, and the it can handle things as it sees fit.